cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -172,40 +166,28 @@
   }
 
   /**
-   * Grants the specified {@link Action actions} on the specified {@link EntityId} to the specified {@link Role} as the
-   * {@link #sentryAdminGroup}. This is used to grant privileges to the dot role created for an entity.
+   * Grants the specified {@link Action actions} on the specified {@link co.cask.cdap.proto.security.Authorizable}
+   * to the specified {@link Role}.
    *
-   * @param entityId the entity on which the actions need to be granted
-   * @param role the role to which the actions need to granted
-   * @param actions the actions which need to be granted
-   * @throws Exception when there is any exception while running the client command to grant
-   */
-  void grant(EntityId entityId, Role role, Set<Action> actions) throws Exception {
-    grant(entityId, role, actions, sentryAdminGroup);
-  }
-
-  /**
-   * Grants the specified {@link Action actions} on the specified {@link EntityId} to the specified {@link Role} as the
-   * {@link #sentryAdminGroup}.
-   *
-   * @param entityId the entity on which the actions need to be granted
+   * @param authorizable the authorizable on which the actions need to be granted
    * @param role the role to which the actions need to granted
    * @param actions the actions which need to be granted
    * @param requestingUser the user executing this operation
    * @throws Exception when there is any exception while running the client command to grant for user
    */
-  void grant(final EntityId entityId, final Role role, Set<Action> actions,
+  void grant(final co.cask.cdap.proto.security.Authorizable authorizable, final Role role, Set<Action> actions,
              final String requestingUser) throws Exception {
     if (!roleExists(role)) {
```

Comment:
After this change we don't need to handle it here it will get wrapped
https://github.com/caskdata/cdap-security-extn/pull/121/files#diff-b1966325d8aac8f6865309f2d055dee8R643